static/app/utils/profiling/profile/sentrySampledProfile.tsx

```diff
@@ -0,0 +1,144 @@
+import {lastOfArray} from 'sentry/utils';
+import {CallTreeNode} from 'sentry/utils/profiling/callTreeNode';
+
+import {Frame} from './../frame';
+import {Profile} from './profile';
+import {createSentrySampleProfileFrameIndex} from './utils';
+
+// This is a simplified port of speedscope's profile with a few simplifications and some removed functionality.
+// head at commit e37f6fa7c38c110205e22081560b99cb89ce885e
+
+// We should try and remove these as we adopt our own profile format and only rely on the sampled format.
+export class SentrySampledProfile extends Profile {
+  static FromProfile(
+    sampledProfile: Profiling.SentrySampledProfile,
+    frameIndex: ReturnType<typeof createSentrySampleProfileFrameIndex>
+  ): Profile {
+    const {samples, stacks, thread_metadata = {}} = sampledProfile.profile;
+    const startedAt = parseInt(samples[0].elapsed_since_start_ns, 10);
+    const endedAt = parseInt(samples[samples.length - 1].elapsed_since_start_ns, 10);
+    const threadId = parseInt(samples[0].thread_id, 10);
+    const threadName = `thread: ${
+      thread_metadata[samples[0].thread_id]?.name || threadId
+    }`;
+    const profileTransactionName = sampledProfile.transactions?.[0]?.name;
+    const profile = new SentrySampledProfile({
+      duration: endedAt - startedAt,
+      startedAt,
+      endedAt,
+      unit: 'nanoseconds',
+      name: profileTransactionName
+        ? `${profileTransactionName} (${threadName})`
+        : threadName,
+      threadId,
+    });
+
+    let previousSampleWeight = 0;
+
+    for (let i = 0; i < samples.length; i++) {
+      const sample = samples[i];
+      const stack = stacks[sample.stack_id];
+      const sampleWeight = parseInt(sample.elapsed_since_start_ns, 10);
+
+      profile.appendSampleWithWeight(
+        stack.map(n => {
+          if (!frameIndex[n]) {
+            throw new Error(`Could not resolve frame ${n} in frame index`);
+          }
+
+          return frameIndex[n];
+        }),
+        sampleWeight - previousSampleWeight
+      );
+
+      previousSampleWeight = sampleWeight;
+    }
+
+    return profile.build();
+  }
+
+  appendSampleWithWeight(stack: Frame[], weight: number): void {
+    // Keep track of discarded samples and ones that may have negative weights
+    this.trackSampleStats(weight);
+
+    // Ignore samples with 0 weight
+    if (weight === 0) {
+      return;
+    }
+
+    let node = this.appendOrderTree;
+    const framesInStack: CallTreeNode[] = [];
+
+    for (const frame of stack) {
```

Comment:
Probably better to just for`(let i = stack.length - 1; i >= 0; i--)` and add a comment why than to call reverse above